docs/book/features/orphan_volumes.md

```diff
@@ -0,0 +1,51 @@
+## Orphan volumes detection and clean-up
+Orphan volumes are [FCDs](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-storage/GUID-AD5AE35E-5209-4775-988C-F86D0E4F0C29.html?hWord=N4IghgNiBcIGYEsBOBnALgAgMYTClGAJgigNYgC+QA) that are present in a vSphere datastore but there is no corresponding PersistentVolume in Kubernetes clusters on the vCenter.
+
+Orphan volumes are often created when CNS solution creates more than one FCD for a Persistent Volume in the Kubernetes cluster. This could occur when the vCenter components are slow, storage is slow, vCenter service restarts, connectivity issues between vCenter and ESXi hosts etc. Since these orphan volumes occupy space in the datastore and are not really used in Kubernetes, it's useful to identify and cleanup orphan volumes periodically.
+
+This functionality provides a set of APIs to detect and delete orphan volumes on-demand, and also provide an option to turn on automatic deletion periodically.
+
+### What qualifies as an orphan volume ?
+An FCD can qualify as an orphan volume if it meets below criteria:
+
+1) A PersistentVolume in any of the registered kubernetes clusters is not using the FCD.
+2) FCD was dynamically provisioned using vSphere CSI driver.
+3) FCD exists for more than 60 minutes(or `vso-create-time-limit-mins` value configured in `cnsmanager-config` configmap) at the time of orphan volumes detection.
+
+### Which orphan volumes are skipped from detection/deletion ?
+Some volumes, even if a kubernetes PersistentVolume doesn't map to them, will not be considered during orphan volume detection/deletion. These include:
+* FCDs created out of band (not using vSphere CSI driver).
+* Statically provisioned CNS volume whose name doesn't start with `pvc-`.
+* File volumes.
+* Orphan volumes that contain snapshots will be detected as orphans but they can not be deleted using Orphan volume delete API.
```

Comment:
This sounds like a limitation of the implementation. Why do we skip?

Reply:
It's a validation on the FCD side which doesn't allow deleting volumes with snapshots. From CNS manager, we still try deleting them but FCD will return an error for these volumes.  

Btw we have also enhanced delete orphan volume API response and it returns which volumes were deleted, which failed to delete along with the reason why it wasn't deleted. Given this, do you think we should remove the note about orphan volumes with snapshots?

Reply:
I'm not sure why FCD does not allow deletion when there are snapshots. At least in this orphan volume case, that does not look right. Do we have an ask on FCD to allow volume deletion when there are snapshots?

Reply:
FCD wouldn't know if a volume is orphan or not. From FCD's perspective we're just invoking the delete FCD API, so won't it make sense to have such a validation?  We currently don't have any ask on FCD, but this use case will be revisited when we add orphan snapshot deletion support in CNS manager.